### PR TITLE
io type fixes; pending a few further updates

### PR DESCRIPTION
**Pull request by cryptodeal**
A few areas where further refinement is required (namely specifying types for route functions & generic for statistics). I've left comments in the code where I intend to go back and do some additional work tomorrow (but if y'all are good with the changes as is, we can merge this and submit a separate PR for those additional changes).

### CODE REVIEW

File: shumai/io/network.ts [hunk 1]
```diff
@@ -78,32 +80,47 @@ export async function tfetch(
...
   const sub_stat_fn = request_dict.statistics.bind({})
-  request_dict.statistics = async (u) => {
+  request_dict.statistics = async (u: sm.Tensor) => {
     if (sub_stat_fn) {
       const s = await sub_stat_fn(u)
-      return {...s, ...statistics }
+      return { ...s, ...statistics }
     }
-    return statistics
+    return { ...statistics }
```
Comment (bwasti) [inline shumai/io/network.ts:120]: why expand here?

File: shumai/io/network.ts [hunk 2]
```diff
@@ -78,32 +80,47 @@ export async function tfetch(
...
   }
-  const serve_request = async (req: Request, fn) => {
+  /* TODO: specify a bettery type than any as its a function */
```
Comment (bwasti) [inline shumai/io/network.ts:122]: nit `s/bettery/better` :)

**Review by bwasti — APPROVED** [review]
LGTM! small nits then I'll merge

File: shumai/io/network.ts (hunk 1 above)
Comment (asilvas-godaddy) [reply shumai/io/network.ts:120]: Not seeing the value in this clone

Comment (cryptodeal) [reply shumai/io/network.ts:120]: Yea, this was honestly a brain fart RE working on this a bit later than intended. Will fix in the next commit!

File: shumai/io/network.ts (hunk 2 above)
Comment (cryptodeal) [reply shumai/io/network.ts:122]: Haha along the same lines as above, fixing now ^^

**Pull request merged**